actionview/lib/action_view/helpers/form_helper.rb

```diff
@@ -1743,7 +1743,9 @@ def id
       # element, sharing a common <tt>id</tt> root (<tt>post_title</tt>, in this
       # case).
       def field_id(method, *suffixes, index: @index)
-        @template.field_id(@object_name, method, *suffixes, index: index)
+        object_name = [@options[:namespace], @object_name].compact_blank.join("_")
```

Comment:
I noticed that in 59ca21c0119fff803e75e78d800c0c98aee7fd4c we wrote [a little extra code to reduce allocations](https://github.com/rails/rails/commit/59ca21c0119fff803e75e78d800c0c98aee7fd4c#diff-35ff9932cd62aff4f22fa9fb8fc9ea39005108d86e8854c8d7a9567cc3871bc7R106-R115).  Would that not be a concern here too?  Perhaps something like:

```suggestion
        object_name = @options[:namespace] ? "#{@options[:namespace]}_#{@object_name}" : @object_name
```